### PR TITLE
Set owner ref for cleanups

### PR DESCRIPTION
**Pull request by arnongilboa**

### CODE REVIEW

File: pkg/internal/checkup/checkup.go [hunk 1]
```diff
@@ -804,6 +807,12 @@ func (c *Checkup) checkVMIHotplugVolume(ctx context.Context, errStr *string) err
 	dv := &cdiv1.DataVolume{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: hotplugVolumeName,
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion: "v1",
+				Kind:       "Pod",
+				Name:       c.checkupConfig.PodName,
```
Comment (ShellyKa13) [inline pkg/internal/checkup/checkup.go:813]: can't podname be empty?

**Review by ShellyKa13 — COMMENTED** [review]
Hey some questions

File: pkg/internal/launcher/launcher.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func (l Launcher) Run(ctx context.Context) (runErr error) {
 	}
 
 	defer func() {
-		if err := l.checkup.Teardown(ctx); err != nil {
+		if err := l.checkup.Teardown(context.Background()); err != nil {
```
[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/internal/launcher/launcher.go:77]: I agree it's not related to the owner ref but to the cleanup on timeout. Before this change, if the passed Run() context timed out, the Teardown() would time out immediately, so as we want to do the cleanup in the timeout case, the context was changed accordingly.

File: pkg/internal/config/config.go [hunk 3]
```diff
@@ -20,31 +20,47 @@
...
 
+	podName := rawEnv[kconfig.PodNameEnvVarName]
+	podUID := rawEnv[kconfig.PodUIDEnvVarName]
+	if podName != "" && podUID == "" {
+		pod, err := client.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+		if err != nil {
+			return kconfig.Config{}, err
+		}
+		rawEnv[kconfig.PodUIDEnvVarName] = string(pod.UID)
+	}
```
[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/internal/config/config.go:62]: kiagnose validates it for us
https://github.com/kiagnose/kubevirt-storage-checkup/blob/2ef9411ddb796ec31d3a18dedea315d83e5b74aa/vendor/github.com/kiagnose/kiagnose/kiagnose/config/environment.go#L62-L64

File: pkg/internal/checkup/checkup.go (hunk 1 above)
Comment (arnongilboa) [reply pkg/internal/checkup/checkup.go:813]: see previous answer

File: pkg/internal/launcher/launcher.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShellyKa13) [reply pkg/internal/launcher/launcher.go:77]: so maybe worth a commit with this  exact explanation :)

[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/internal/launcher/launcher.go:77]: done

File: pkg/internal/config/config_test.go [hunk 4]
```diff
@@ -38,31 +38,33 @@ import (
...
+	testPodName       = "pod"
+	testPodUID        = "uid"
 )
 
 var testEnv = map[string]string{
 	kconfig.ConfigMapNamespaceEnvVarName: testNamespace,
 	kconfig.ConfigMapNameEnvVarName:      testConfigMapName,
 	kconfig.PodNameEnvVarName:            testPodName,
+	kconfig.PodUIDEnvVarName:             testPodUID,
 }
```
[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/internal/config/config_test.go:51]: Sure, added.

**Pull request merged**